connectors/spark/src/main/scala/io/unitycatalog/spark/utils/Clock.java

```diff
@@ -0,0 +1,55 @@
+package io.unitycatalog.spark.utils;
+
+import java.time.Duration;
+import java.time.Instant;
+
+public interface Clock {
+  /**
+   * @return the current time of the clock.
+   */
+  Instant now();
+
+  /**
+   * Advances the current time of this clock by the specified duration. After this call,
+   * {@link #now()} should return a time equal to the previous time plus the given {@code duration}.
+   */
+  void advance(Duration duration);
+
+  static Clock systemClock() {
+    return new SystemClock();
```

Comment:
minor: this does not need to return new object every time. this can be a singleton.